api/envoy/type/tracing.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+package envoy.type;
+
+option java_outer_classname = "TracingProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Tracing]
+
+// Describes custom tags for the active span.
+message TracingCustomTag {
+  // Literal type custom tag with literal value for tag value.
+  message Literal {
+    // literal value to populate the tag value.
+    string value = 1 [(validate.rules).string = {min_bytes: 1}];
+  }
+
+  // Environment type custom tag with environment name and default value.
+  message Environment {
+    // environment variable name to obtain the value to populate the tag value.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // used to populate the tag value is the environment variable is not found.
+    string default_value = 2;
+  }
+
+  // RequestHeader type custom tag with header name and default value.
+  message RequestHeader {
```

Comment:
We also need `RequestMetadata` here.
We would like to add specific metadata as custom tags. For Istio this would be the source configuration artifacts that resulted in creating the matched route.

Reply:
would you have a look about the new `RequestMetadata`.